### PR TITLE
ci: add containerd PR E2E tests

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR adds containerd configuration to the PR E2E jobs, so that we test containerd scenarios before merging changes.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: examples/e2e-tests/kubernetes/release/default/definition.json [hunk 1]
```diff
@@ -121,15 +121,6 @@
         "distro": "ubuntu-18.04-gen2",
         "availabilityProfile": "VirtualMachineScaleSets",
         "vnetSubnetId": "/subscriptions/SUB_ID/resourceGroups/RG_NAME/providers/Microsoft.Network/virtualNetworks/VNET_NAME/subnets/SUBNET_NAME"
-      },
-      {
```
Comment (jackfrancis) [inline examples/e2e-tests/kubernetes/release/default/definition.json:125]: I don't see any reason to include two Windows pools on the standard PR E2E config

Comment (marosset) [reply examples/e2e-tests/kubernetes/release/default/definition.json:125]: me either

File: test/e2e/engine/template.go [hunk 2]
```diff
@@ -231,9 +231,19 @@ func Build(cfg *config.Config, masterSubnetID string, agentSubnetIDs []string, i
...
+		if prop.OrchestratorProfile.KubernetesConfig.WindowsContainerdURL == "" {
+			prop.OrchestratorProfile.KubernetesConfig.WindowsContainerdURL = "https://github.com/containerd/containerd/releases/download/v1.4.0/containerd-1.4.0-windows-amd64.tar.gz"
+		}
+		if prop.WindowsProfile.WindowsPublisher == "" &&
+			prop.WindowsProfile.WindowsOffer == "" &&
+			prop.WindowsProfile.WindowsSku == "" &&
+			prop.WindowsProfile.ImageVersion == "" {
+			prop.WindowsProfile.WindowsPublisher = "microsoft-aks"
+			prop.WindowsProfile.WindowsOffer = "aks-windows"
+			prop.WindowsProfile.WindowsSku = "2019-datacenter-core-smalldisk-containerd-2005"
```
Comment (marosset) [inline test/e2e/engine/template.go:244]: I'm torn between thinking we should publish containerd images monthly like we publish the docker images and not wanting to have to publish any more images to the marketplace than necessary...

**Review by marosset — APPROVED** [review]
/lgtm

**Pull request merged**